src/accelerate/utils/operations.py

```diff
@@ -100,7 +115,7 @@ def recursively_apply(func, data, *args, test_type=is_torch_tensor, error_on_oth
     elif test_type(data):
         return func(data, *args, **kwargs)
     elif error_on_other_type:
-        raise ValueError(
+        raise TypeError(
             f"Can't apply {func.__name__} on object of type {type(data)}, only of nested list/tuple/dicts of objects "
             f"that satisfy {test_type.__name__}."
```

Comment:
while at it could this message be reworded to make sense "only of nested" can't be parsed, IMHO.

Reply:
perhaps, `"Expected all elements of list/tuple/dict to be a tensor, but got {type(data)}"`

Reply:
I'll do this in a follow up PR as it's in more than one place.